internal/services/cognitive/cognitive_account_customer_managed_key_resource_test.go

```diff
@@ -0,0 +1,231 @@
+  cognitive_account_id = azurerm_cognitive_account.test.id
+  key_vault_key_id     = azurerm_key_vault_key.test.id
+  identity_client_id   = azurerm_user_assigned_identity.test.client_id
+}
+`, r.template(data))
+}
+
+func (r CognitiveAccountCustomerManagedKeyResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+    key_vault {
+      purge_soft_delete_on_destroy = false
+    }
+  }
+}
+
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-cognitive-%d"
+  location = "%s"
+}
+
+resource "azurerm_user_assigned_identity" "test" {
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  name                = "%s"
+}
+
+resource "azurerm_cognitive_account" "test" {
+  name                  = "acctest-cogacc-%d"
+  location              = azurerm_resource_group.test.location
+  resource_group_name   = azurerm_resource_group.test.name
+  kind                  = "Face"
+  sku_name              = "E0"
+  custom_subdomain_name = "acctest-cogacc-%d"
+  identity {
+    type         = "SystemAssigned, UserAssigned"
+    identity_ids = [azurerm_user_assigned_identity.test.id]
+  }
+}
+
+resource "azurerm_key_vault" "test" {
+  name                     = "acctestkv%s"
+  location                 = azurerm_resource_group.test.location
+  resource_group_name      = azurerm_resource_group.test.name
+  tenant_id                = data.azurerm_client_config.current.tenant_id
+  sku_name                 = "standard"
+  soft_delete_enabled      = true
+  purge_protection_enabled = true
+}
+
+resource "azurerm_key_vault_access_policy" "test" {
+  key_vault_id       = azurerm_key_vault.test.id
+  tenant_id          = azurerm_cognitive_account.test.identity.0.tenant_id
+  object_id          = azurerm_cognitive_account.test.identity.0.principal_id
+  key_permissions    = ["get", "create", "list", "restore", "recover", "unwrapKey", "wrapKey", "purge", "encrypt", "decrypt", "sign", "verify"]
+  secret_permissions = ["get"]
+}
+
+resource "azurerm_key_vault_access_policy" "test2" {
+  key_vault_id       = azurerm_key_vault.test.id
+  tenant_id          = data.azurerm_client_config.current.tenant_id
+  object_id          = data.azurerm_client_config.current.object_id
+  key_permissions    = ["get", "create", "delete", "list", "restore", "recover", "unwrapKey", "wrapKey", "purge", "encrypt", "decrypt", "sign", "verify"]
+  secret_permissions = ["get"]
+}
+
+resource "azurerm_key_vault_access_policy" "test3" {
+  key_vault_id       = azurerm_key_vault.test.id
+  tenant_id          = azurerm_user_assigned_identity.test.tenant_id
+  object_id          = azurerm_user_assigned_identity.test.principal_id
+  key_permissions    = ["get", "create", "delete", "list", "restore", "recover", "unwrapKey", "wrapKey", "purge", "encrypt", "decrypt", "sign", "verify"]
+  secret_permissions = ["get"]
+}
```

Comment:
fwiw since the `user_assigned_identity` is being created separately, these 3 could be inlined within the `azurerm_key_vault` resource which'd remove the `depends_on` below